File: models/developing/pml_examples/pml1d/test1d.m
% Driver for simple PML test: block sitting on a half space.
% Version 1: Use Lua for everything.

% HiQLab
% Copyright (c): Regents of the University of California
% $Id: test1d.m,v 1.1 2006/05/01 08:20:34 tkoyama Exp $

% -- Assemble and solve linear system

param.k     = 5;
param.order = 3;
param.w     = 1;

mesh  = Mesh_load('test1d.lua', param);
Mesh_make_harmonic(mesh, param.w);
[M,K] = Mesh_assemble_mk(mesh);
F     =-Mesh_assemble_R(mesh);
u     = (K-param.w^2*M)\F;
Mesh_set_u(mesh, u);

% -- Plot forced response

opt.axis = [0, 12, -1, 1];
plotcycle1d(mesh, 1, opt);
Mesh_delete(mesh);

File: sandbox/parallel/petsc/test/test1d.lua
-- 1D test mesh

-- HiQLab
-- Copyright (c): Regents of the University of California
 
-- Set up properties

order = order or 3    -- Polynomial order
nelt  = nelt or 1000  -- Number of elements

-- Assemble mesh quantities

mesh = Mesh:new(1);
D1   = mesh:PMLScalar1d(1, 1);
mesh:add_block(0, 1, nelt*order+1, D1, order);

-- Define BC

mesh:set_bc(function(x)  
  if x == 0          then return 'u', 1 end  
  if abs(x-1) < 1e-8 then return 'u', 0 end
end)

